gems/aws-sdk-core/lib/aws-sdk-core/plugins/endpoint_discovery.rb

```diff
@@ -4,10 +4,10 @@ module Plugins
     class EndpointDiscovery < Seahorse::Client::Plugin
 
       option(:endpoint_discovery,
-        default: false,
+        doc_default: Proc.new { |options| options[:endpoint_discovery_required] },
```

Comment:
Why is this a Proc?

Reply:
To allow it to be called dynamically during code generation